File: platform/frontend/src/api/adapters/management.ts
import type {
  CaseSummary,
  EnvironmentDetail,
  EnvironmentSummary,
  ScheduleSummary,
  SuiteSummary,
} from '@/types/platform';

type UnknownRecord = Record<string, unknown>;

const isRecord = (value: unknown): value is UnknownRecord =>
  typeof value === 'object' && value !== null && !Array.isArray(value);

const readString = (value: unknown) => (typeof value === 'string' ? value : '');
const readNumber = (value: unknown) => (typeof value === 'number' && Number.isFinite(value) ? value : 0);

export const extractManagementCollection = (payload: unknown): unknown[] => {
  if (Array.isArray(payload)) {
    return payload;
  }

  if (!isRecord(payload)) {
    return [];
  }

  const candidates = ['items', 'data', 'results', 'records', 'list'] as const;
  for (const key of candidates) {
    const value = payload[key];

    if (Array.isArray(value)) {
      return value;
    }
  }

  return [];
};

export const normalizeCaseSummary = (payload: unknown, index: number): CaseSummary => {
  if (!isRecord(payload)) {
    return {
      id: `case-${index}`,
      name: '未命名用例',
      module: '未分类',
      method: 'GET',
      priority: 'P2',
      status: 'draft',
    };
  }

  const method = readString(payload.method).toUpperCase();
  const priority = readString(payload.priority).toUpperCase();
  const status = readString(payload.status).toLowerCase();

  return {
    id: readString(payload.id) || readString(payload.case_id) || `case-${index}`,
    name: readString(payload.name) || readString(payload.title) || '未命名用例',
    module: readString(payload.module) || readString(payload.domain) || '未分类',
    method:
      method === 'GET' || method === 'POST' || method === 'PUT' || method === 'DELETE'
        ? method
        : 'GET',
    priority: priority === 'P0' || priority === 'P1' || priority === 'P2' ? priority : 'P2',
    status: status === 'active' || status === 'draft' ? status : 'draft',
  };
};

export const normalizeSuiteSummary = (payload: unknown, index: number): SuiteSummary => {
  if (!isRecord(payload)) {
    return {
      id: `suite-${index}`,
      name: '未命名套件',
      caseCount: 0,
      lastRun: '尚未执行',
      schedule: '未配置',
    };
  }

  return {
    id: readString(payload.id) || readString(payload.suite_id) || `suite-${index}`,
    name: readString(payload.name) || readString(payload.title) || '未命名套件',
    caseCount: readNumber(payload.caseCount ?? payload.case_count),
    lastRun: readString(payload.lastRun) || readString(payload.last_run) || '尚未执行',
    schedule: readString(payload.schedule) || '未配置',
  };
};

export const normalizeEnvironmentSummary = (
  payload: unknown,
  index: number,
): EnvironmentSummary => {
  if (!isRecord(payload)) {
    return {
      id: `environment-${index}`,
      name: '未命名环境',
      baseUrl: '',
      authMode: '未配置',
      status: 'draft',
    };
  }

  const status = readString(payload.status).toLowerCase();
  return {
    id: readString(payload.id) || readString(payload.environment_id) || `environment-${index}`,
    name: readString(payload.name) || '未命名环境',
    baseUrl: readString(payload.baseUrl) || readString(payload.base_url),
    authMode: readString(payload.authMode) || readString(payload.auth_mode) || '未配置',
    status: status === 'online' || status === 'draft' ? status : 'draft',
  };
};

export const normalizeEnvironmentDetail = (payload: unknown, index: number): EnvironmentDetail => {
  const summary = normalizeEnvironmentSummary(payload, index);
  const variables =
    isRecord(payload) && isRecord(payload.variables)
      ? payload.variables
      : {};

  return {
    ...summary,
    variables,
  };
};

export const normalizeScheduleSummary = (payload: unknown, index: number): ScheduleSummary => {
  if (!isRecord(payload)) {
    return {
      id: `schedule-${index}`,
      name: '未命名任务',
      cron: '',
      target: '未配置目标',
      lastRun: '尚未执行',
      status: 'paused',
    };
  }

  const status = readString(payload.status).toLowerCase();
  return {
    id: readString(payload.id) || readString(payload.schedule_id) || `schedule-${index}`,
    name: readString(payload.name) || '未命名任务',
    cron: readString(payload.cron) || readString(payload.cron_expression),
    target: readString(payload.target) || readString(payload.target_suite_id) || '未配置目标',
    environmentId: readString(payload.environmentId) || readString(payload.environment_id) || undefined,
    environmentLabel:
      readString(payload.environmentLabel) || readString(payload.environment_name) || undefined,
    lastRun: readString(payload.lastRun) || readString(payload.last_run) || '尚未执行',
    status: status === 'active' || status === 'paused' ? status : 'paused',
  };
};

File: platform/backend/src/pyta_platform_backend/schemas/management.py
"""平台管理面 schema。

第一阶段先给管理页提供最小可用的数据契约：
- cases: 用例目录摘要
- suites: 套件目录摘要
- environments: 环境目录摘要
- schedules: 调度目录摘要

这些接口当前主要服务页面展示和 environment 的首个完整资源闭环，
暂不承担完整的复杂编辑能力。
"""

from typing import Dict, Optional

from pydantic import BaseModel, Field


class CaseSummaryResponse(BaseModel):
    """用例列表项。"""

    id: str
    name: str
    module: str
    method: str
    priority: str
    status: str


class SuiteSummaryResponse(BaseModel):
    """套件列表项。"""

    id: str
    name: str
    case_count: int
    last_run: str = Field(..., description="最近一次 run 的展示时间；没有记录时返回提示文案")
    schedule: str


class EnvironmentSummaryResponse(BaseModel):
    """环境列表项。"""

    id: str
    name: str
    base_url: str
    auth_mode: str
    status: str


class EnvironmentDetailResponse(EnvironmentSummaryResponse):
    """环境详情。

    当前先把 `variables` 透传出来，后续前端真正接变量编辑器时，
    就不需要再重新发明一套详情结构。
    """

    variables: Dict[str, object] = Field(default_factory=dict)


class CreateEnvironmentRequest(BaseModel):
    """创建环境的最小请求体。"""

    name: str = Field(..., description="环境名称")
    base_url: str = Field(..., description="环境基础地址")
    auth_mode: str = Field(..., description="鉴权模式说明")
    status: str = Field("draft", description="环境状态：online / draft")
    variables: Dict[str, object] = Field(default_factory=dict, description="预留：环境变量")


class CreateEnvironmentResponse(EnvironmentDetailResponse):
    """创建环境后的确认信息。"""


class UpdateEnvironmentRequest(BaseModel):
    """更新环境的 patch 请求体。"""

    name: Optional[str] = Field(None, description="环境名称")
    base_url: Optional[str] = Field(None, description="环境基础地址")
    auth_mode: Optional[str] = Field(None, description="鉴权模式说明")
    status: Optional[str] = Field(None, description="环境状态：online / draft")
    variables: Optional[Dict[str, object]] = Field(None, description="可选：环境变量")


class UpdateEnvironmentResponse(EnvironmentDetailResponse):
    """更新环境后的响应。"""


class DeleteEnvironmentResponse(EnvironmentSummaryResponse):
    """删除环境后的确认响应。"""


class ScheduleSummaryResponse(BaseModel):
    """调度列表项。"""

    id: str
    name: str
    cron: str
    target: str
    environment_id: Optional[str] = None
    environment_name: Optional[str] = None
    last_run: str = Field(..., description="最近一次与目标套件相关的 run 时间")
    status: str
